app/components/newsletter-item/edit-panel.js

```diff
@@ -63,7 +63,29 @@ export default class NewsletterItemEditPanelComponent extends Component {
     this.newsletterItem.remark = this.remark;
     this.newsletterItem.finished = this.isFinished;
     this.newsletterItem.themes = this.selectedThemes;
-    this.newsletterItem.richtext = this.editorInstance.htmlContent;
+    try {
+      // The editor introduces &nbsp; instead of normal spaces to work around
+      // certain browsers' behavior where normal spaces on outer ends of text nodes
+      // aren't rendered in the content-editable. In recent versions of the editor however,
+      // this &nbsp;-inserting seems to happen too often, which results in very long
+      // lines that don't break, which is undesired.
+      // Here we replace all &nbsp's that don't lean against html tags in an attempt
+      // to keep the editor's workaround behavior, while replacing unnecessary &nbsp;'s
+      //
+      // See KAS-3598 for some extra context. This workaround was removed
+      // becuase at first it didn't seem to apply anymore. But then we found
+      // that the erroneous &nbsp; insertion happened when writing in the middle
+      // of a line. That is, when writing in the editor from a clean slate,
+      // normal spaces were inserted. But when writing in the middle of such a
+      // sentence, all new spaces would become non-breaking spaces.
+      // See also: https://github.com/lblod/ember-rdfa-editor/issues/317
+      //
+      const richtext = this.editorInstance.htmlContent;
+      const cleanedHtml = richtext.replaceAll(/(?<!>)&nbsp;(?!<)/gm, ' ');
+      this.newsletterItem.richtext = cleanedHtml;
+    } catch {
+      // pass
+    }
```

Comment:
Kan dit falen? WIllen we dan de gebruiker niet inlichten?
Anders hebben we mogelijke "verdwijnende kort-bestek berichten"

Reply:
Zou dan de fallback moeten zijn dat we gewoon `this.newsletterItem.richtext = this.editorInstance.htmlContent;` doen?

Reply:
Geen idee. Afaik is het enige dat hier kan falen dat de negative lookbehind regex assertion niet gesupport is door de browser. Maar dat zou niet (meer) het geval mogen zijn, want alle moderne versies van Firefox en Chrome supporten het. Safari heeft daar geen support voor, dus mogelijks is de try-catch daarvoor geintroduceerd, voor zij die Kaleidos soms openen in Safari (Johan denk ik?)

In any case, de code is gewoon klakkeloos overgenomen, dit is de workaround die ervoor hier was, met een kleine toevoeging in comments.

Reply:
Het is een code restore, op zich ok.
Maar nu we erop stoten mogelijk best ineens een fallback voorzien, moest die replaceAll ooit toch falen?

Reply:
Die `catch` is daar precies geintroduceerd om de text value niet te overschrijven met een lege string als de `editorInstance` unavailable is. Waar, hoe of waarom het nu nog kan gebeuren, geen idee.

https://github.com/kanselarij-vlaanderen/frontend-kaleidos/commit/75f1521395820f43127d8b7c6c13536e97c85f17

Reply:
cava voorlopig, later te bekijken of die catch ooit bereikt wordt

Reply:
idd. Lokaal voert hij hier in ieder geval de cleanedHtml goed uit